ext/opcache/jit/zend_jit_trace.c

```diff
@@ -4117,6 +4117,14 @@ static const void *zend_jit_trace(zend_jit_trace_rec *trace_buffer, uint32_t par
 
 	checkpoint = zend_arena_checkpoint(CG(arena));
 
+	zend_accel_hash_entry *accel_h_entry = zend_accel_hash_find_entry(&ZCSG(hash), trace_buffer->op_array->filename);
```

Comment:
Possibly this can fetch a different version of the script (e.g. if the file was discarded and recompiled).

An alternative would be to store the script into `zend_jit_op_array_trace_extension`. (After `op_array`, as I believe the `func_info` and `op_array` fields must be at the same offset in all `zend_jit_*_extension` structs.)